warehouse/query-core/src/main/java/datawave/query/jexl/JexlASTHelper.java

```diff
@@ -369,11 +369,9 @@ public static String getIdentifier(JexlNode node, boolean deconstruct) throws No
                     for (int j = 0; j < child.jjtGetNumChildren(); j++) {
                         JexlNode grandChild = child.jjtGetChild(j);
                         
-                        // If the grandchild and its image is non-null and equal to the any-field identifier
+                        // if the grandChild is an identifier, then return its image/value
                         if (grandChild instanceof ASTIdentifier) {
                             return (deconstruct ? deconstructIdentifier(grandChild.image) : grandChild.image);
-                        } else if (grandChild instanceof ASTFunctionNode) {
-                            return null;
```

Comment:
Intellij makes the valid argument that the top level for loop in this method doesn't loop at all.  If the child is a reference, we loop over it's children and either return an identifier or null.  If the child is anything else, we immediately return null.  Should we rework this method a little more?

Reply:
Sometimes I wonder how this works at all....wow